src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -422,20 +420,24 @@ - (void)writeAttributeWithEndpointID:(NSNumber *)endpointID
                expectedValueInterval:(NSNumber *)expectedValueInterval
                    timedWriteTimeout:(NSNumber * _Nullable)timeout
 {
-    // Start the asynchronous operation
-    MTRBaseDevice * baseDevice = [self newBaseDevice];
-    [baseDevice
-        writeAttributeWithEndpointID:endpointID
-                           clusterID:clusterID
-                         attributeID:attributeID
-                               value:value
-                   timedWriteTimeout:timeout
-                               queue:self.queue
-                          completion:^(NSArray<NSDictionary<NSString *, id> *> * _Nullable values, NSError * _Nullable error) {
-                              if (values) {
-                                  [self _handleAttributeReport:values];
-                              }
-                          }];
+    MTRAsyncCallbackQueueWorkItem * workItem = [[MTRAsyncCallbackQueueWorkItem alloc] initWithQueue:_queue];
+    MTRAsyncCallbackReadyHandler readyHandler = ^(MTRDevice * device, NSUInteger retryCount) {
+        MTRBaseDevice * baseDevice = [self newBaseDevice];
```

Comment:
This is going to need to be merged on top of https://github.com/project-chip/connectedhomeip/pull/23155 ...

Probably simplest is to re-add `newBaseDevice` in this PR or something, and just put it in the same place where it used to be, so when we try to re-land the API change PRs the conflicts are minimal.  But the selector on the baseDevice call will also need updating, unless we re-land some of the API bits first.

Reply:
Done